lib/http.js

```diff
@@ -23,11 +28,22 @@ const client = require('_http_client');
 const ClientRequest = exports.ClientRequest = client.ClientRequest;
 
 exports.request = function request(options, cb) {
+  if (typeof options === 'string') {
+    options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
+  } else if (options instanceof url.URL) {
```

Comment:
@mscdex, I didn't want to duplicate the `options` check too, but w/o the tests with `options` of type `string` fail as `_defaultAgent` cannot be added as a property on it. Instead of this, should I just check if type of `options` is `Object` or not? Is there another workaround?